pkgs/test/lib/src/runner/browser/dom.dart

```diff
@@ -2,29 +2,74 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:js_interop';
 import 'dart:js_util' as js_util;
 
 import 'package:js/js.dart';
 
+// Conversion functions to help with the migration.
+extension ObjectToJSAnyExtension on Object? {
+  JSAny? get toJSAnyShallow {
+    if (const bool.fromEnvironment('dart.library.html')) {
```

Comment:
is this basically detecting wasm vs js? Maybe add a comment to that effect just to clarify (my initial reaction was this wasn't necessary, because we are always running on the web, but wasm is also web which is easy to forget :P ).